pkg/services/alertmanager.go

```diff
@@ -0,0 +1,207 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	texttemplate "text/template"
+	"time"
+
+	httputil "github.com/argoproj/notifications-engine/pkg/util/http"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	alertNameLabel = "alertname"
+)
+
+// AlertmanagerNotification message body is similar to Prometheus alertmanager postableAlert model
+type AlertmanagerNotification struct {
+	Labels       map[string]string `json:"labels"`
+	Annotations  map[string]string `json:"annotations"`
+	GeneratorURL string            `json:"generatorURL"`
+	StartsAt     time.Time         `json:"startsAt"`
```

Comment:
I have read the doc of Alertmanager.
https://prometheus.io/docs/alerting/latest/clients/

I guess the fields of startsAt and endsAt are optional, so it is good that they can set EndsAt like using interval as `15m`.
I mean if they are setting below config, the request will be set the fields for `EndsAt` and it is `time.Now() + 15m`.

```
alertmanager:
    interval: 15m
    labels:
      fault_priority: "P5"
      event_bucket: "deploy"
      event_status: "succeed"
      recipient: "{{.recipient}}"
```

What do you think?

Reply:
`Endsat` is used to make alertmanager generate a recovered notification, but this notification is a little unnecessary for argocd. I specifically stated in the [dcos](https://github.com/mingqing/notifications-engine/blob/feature-alertmanager/docs/services/alertmanager.md#prometheus-alertmanager-config) that it needs to be disabled.